File: 3rd Sem/NM/python/trapezoidalrule.py
n = int(input("Enter number of intervals: "))
a, b = map(float, input("Enter lower and upper limit: ").split())

def f(x): 
    return x**2  # The function to integrate (x^2)

h = (b - a) / n  # Step size

# Initializing sum with f(a) + f(b)
s = f(a) + f(b)

# Summing up the middle terms (2 * f(x))
for i in range(1, n):
    s += 2 * f(a + i * h)

# Calculating the integral using the Trapezoidal Rule
integral = (h / 2) * s

print("Integral:", integral)

File: 3rd Sem/NM/C-program/trapezoidalrule.c
#include <stdio.h>

double f(double x) {
    return x * x;
}

int main() {
    int n;
    double a, b;
    
    printf("Enter number of intervals: ");
    scanf("%d", &n);
    
    printf("Enter lower and upper limit: ");
    scanf("%lf %lf", &a, &b);
    
    double h = (b - a) / n;
    double s = f(a) + f(b);
    
    for (int i = 1; i < n; i++) {
        s += 2 * f(a + i * h);
    }
    
    printf("Integral: %.6f\n", (h / 2) * s);
    
    return 0;
}
